hibernate-rx-core/src/test/java/org/hibernate/rx/configuration/RxConnectionPoolTest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.rx.configuration;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.rx.cfg.AvailableRxSettings;
+import org.hibernate.rx.containers.PostgreSQLDatabase;
+import org.hibernate.rx.service.RxConnectionPoolProviderImpl;
+import org.hibernate.rx.service.initiator.RxConnectionPoolProvider;
+import org.junit.Test;
+
+import io.vertx.pgclient.PgPool;
+import io.vertx.sqlclient.Pool;
+
+public class RxConnectionPoolTest {
```

Comment:
Could you add a test to make sure that the right exception is thrown when the Object is of the wrong type?
Also what happens  if the credentials or something else is wrong

Reply:
Good idea, I'll add a test for invalid Vertx pool type.

As for invalid credentials, that would get passed through and when the first connection is made we'd get the error/message directly from the Vertx driver.